WordPress/openverse
# Drop and recreate matview in data refresh

## Fixes



## Description



This PR updates the data refresh DAGs to drop and recreate the materialized view, rather than refreshing it. This is done because refreshing the view is timing out after 21 days in production, but dropping and recreating succeeds in ~5 hours.

This change is _temporary_. #1894 describes a plan for decoupling the popularity steps from the data refresh: as part of that work, these steps will be removed entirely from the data refresh DAG and refactored. However these changes will allow us to keep the image data refresh enabled while that work continues.

The data refresh DAG (note that `recreate_matview` is now a TaskGroup):
<img width="1226" alt="Screen Shot 2023-05-03 at 12 07 11 PM" src="https://user-images.githubusercontent.com/63313398/236018844-096ef2fc-530b-4a98-9338-41e10e796ddf.png">

The taskgroup expanded:
<img width="1321" alt="Screen Shot 2023-05-03 at 12 07 56 PM" src="https://user-images.githubusercontent.com/63313398/236019057-a22e9264-c09f-4296-84a9-87dd985648c1.png">


## Testing Instructions

Try running the image and audio data refreshes locally and ensure they pass. Run each twice; assuming you have not run the refresh previously this month, the first run should refresh the popularity metrics and constants before recreating the matview, and the second should skip these steps and proceed directly to recreating the matview.





## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
